File: client-cache-db/app/src/main/java/ua/meugen/android/client/di/AppComponent.java
package ua.meugen.android.client.di;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import ua.meugen.android.client.Client;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent extends AndroidInjector<Client> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<Client> {}
}

File: client-coroutines/app/src/main/java/ua/meugen/android/client/coroutines/app/di/AppComponent.kt
package ua.meugen.android.client.coroutines.app.di

import dagger.Component
import dagger.android.AndroidInjector
import ua.meugen.android.client.coroutines.app.ClientCoroutines
import ua.meugen.android.client.coroutines.app.di.modules.AppModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent: AndroidInjector<ClientCoroutines> {

    @Component.Builder
    abstract class Builder: AndroidInjector.Builder<ClientCoroutines>() {}
}